sel4-ftrace-merge/include/ftrace.h

```diff
@@ -0,0 +1,213 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/*
+ * Copyright 2023, Technology Innovation Institute
+ *
+ */
+#ifndef FTRACE_H
```

Comment:
I know we are not inside seL4 here, but seL4 coding guidelines dictate` #pragma once` should be used instead. The reason I'm bringing this up, because I know some standards dictate the naming scheme of the preprocessor identifier used. I would think nowadays compilers support `#pragma once` and this way we would not need to bother ourselves what's the correct naming. Opinions @elmankku ?

Reply:
> I know we are not inside seL4 here, but seL4 coding guidelines dictate` #pragma once` should be used instead. The reason I'm bringing this up, because I know some standards dictate the naming scheme of the preprocessor identifier used. I would think nowadays compilers support `#pragma once` and this way we would not need to bother ourselves what's the correct naming. Opinions @elmankku ?

Yes agreed, `#pragma once` is good default for the new stuff - makes life a bit easier. Naturally we should follow repo/upstream conventions when applicable.

Reply:
What do you think @jsvapiav ? Could you live with `#pragma once`?

Reply:
Sure, NP =)